UltraStar Play/Assets/Common/Playlist/UltraStarPlaylistParser.cs

```diff
@@ -0,0 +1,126 @@
+
+    public static UltraStarPlaylist ParseFile(string path)
+    {
+        UltraStarPlaylist playlist = new UltraStarPlaylist();
+        string[] lines = File.ReadAllLines(path);
+        for (int lineIndex = 0; lineIndex < lines.Length; lineIndex++)
+        {
+            try
+            {
+                UltraStartPlaylistLineEntry lineEntry = UltraStarPlaylistLineParser.ParseLine(lines[lineIndex]);
+                playlist.AddLineEntry(lineEntry);
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(new UltraStarPlaylistParserException($"Error in line {lineIndex + 1} in file '{path}'", e));
+            }
+        }
+        return playlist;
+    }
+
+    public static class UltraStarPlaylistLineParser
+    {
+        private enum Token
+        {
+            Artist, Title, Separator
+        }
+
+        public static UltraStartPlaylistLineEntry ParseLine(string line)
+        {
+            if (line.Trim().StartsWith("#"))
+            {
+                // This is a comment
+                return new UltraStartPlaylistLineEntry(line);
+            }
+
+            // Read the line character by character.
+            // Add this character either to the artist or title, depending on the targetToken.
+            Token targetToken = Token.Artist;
+            bool insideQuote = false;
+            string artist = "";
+            string title = "";
+            char lastChar = '0';
+            foreach (char c in line)
+            {
+                if (c == '"' && lastChar != '\\')
+                {
+                    insideQuote = !insideQuote;
+                    if (insideQuote)
+                    {
+                        // What has been read before for this token is now obsolete
+                        if (targetToken == Token.Artist)
+                        {
+                            artist = "";
+                        }
+                        else if (targetToken == Token.Title)
+                        {
+                            title = "";
+                        }
+                    }
+                    else if (targetToken == Token.Artist)
+                    {
+                        // The artist token has been completed, the separator comes next.
+                        targetToken = Token.Separator;
+                    }
+                    else if (targetToken == Token.Title)
+                    {
+                        // The title has been completed, all done.
+                        break;
+                    }
+                }
+                else if (!insideQuote && c == separator
+                    && (targetToken == Token.Artist || targetToken == Token.Separator))
+                {
+                    targetToken = Token.Title;
+                }
+                else if (c != '\\' || (c == '\\' && lastChar == '\\'))
+                {
+                    switch (targetToken)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5574890)